python/ray/tune/tests/test_result_grid.py

```diff
@@ -224,6 +224,39 @@ def f(config):
     assert not any(key in representation for key in AUTO_RESULT_KEYS)
 
 
+def test_result_grid_repr(ray_start_2_cpus, tmpdir):
+    from ray.air import session
+
+    def f(config):
+        if config["x"] == 1:
+            raise RuntimeError
+
+        metrics = {"loss": 1}
+        session.report(metrics, checkpoint=Checkpoint.from_dict(metrics))
+
+    tuner = tune.Tuner(
+        f,
+        run_config=air.RunConfig(
+            name="test_result_grid_repr",
+            local_dir=str(tmpdir / "test_result_grid_repr_results"),
+        ),
+        param_space={"x": tune.grid_search([1, 2])},
+    )
+    result_grid = tuner.fit()
+
+    representation = result_grid.__repr__()
```

Comment:
Can we instead construct the result grid manually and assert the layout?

Reply:
I agree. Manually creating a result grid will be way faster than running a full tuning experiment.